# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1209959195 14400
#      Sun May 04 23:46:35 2008 -0400
# Node ID 45de7d8dac72d28f0d0db2390c7c8e346bcdb4cc
# Parent  ea9cb4d68dbf333ea3ba23297567c0eff51a682b
ov-fcn-handle.cc (Ffunctions): fix structure assignment

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,13 @@
 2008-05-04  John W. Eaton  <jwe@octave.org>
 
+	* ov-fcn-handle.cc (Ffunctions): Pass octave_value object instead
+	of Cell in structure field assignment.
+
 	* parse.y (frob_function): Don't install subfunctions here.
 	(finish_function): Handle subfunctions here.
 	Conditionally define tree_function_def object here.
 
 	* symtab.h (symbol_table::fcn_info::fcn_info_rep::find_function):
 	Initialize args_evaluated.
 	* pt-id.cc (tree_identifier::rvalue): Likewise.
 	* variables.cc (symbol_exist): Likewise.
diff --git a/src/ov-fcn-handle.cc b/src/ov-fcn-handle.cc
--- a/src/ov-fcn-handle.cc
+++ b/src/ov-fcn-handle.cc
@@ -1229,17 +1229,17 @@ Return a struct containing information a
 		  m.assign ("function", fh_nm);
 
 		  if (fcn->is_nested_function ())
 		    {
 		      m.assign ("type", "subfunction");
 		      Cell parentage (dim_vector (1, 2));
 		      parentage.elem(0) = fh_nm;
 		      parentage.elem(1) = fcn->parent_fcn_name ();
-		      m.assign ("parentage", parentage); 
+		      m.assign ("parentage", octave_value (parentage)); 
 		    }
 		  else
 		    m.assign ("type", "simple");
 		}
 
 	      std::string nm = fcn->fcn_file_name ();
 
 	      if (nm.empty ())
